application/usecase/pokemon.go

```diff
@@ -0,0 +1,68 @@
+package usecase
+
+import (
+	"strconv"
+
+	"github.com/FernandoGal25/academy-go-q42021/application/contracts"
+	"github.com/FernandoGal25/academy-go-q42021/domain/model"
+	errors "github.com/FernandoGal25/academy-go-q42021/error"
+)
+
+const TOTAL_POKEMON = 898
+
+/*
+	Service that handles the pokemon case uses.
+*/
+type PokemonService struct {
+	CSVPokemonRepository  contracts.PokemonRepository
+	RestPokemonRepository contracts.PokemonRepository
+}
+
+/*
+	Returns an instance of PokemonService.
+*/
+func NewPokemonService(r1 contracts.PokemonRepository, r2 contracts.PokemonRepository) PokemonService {
+	return PokemonService{r1, r2}
+}
+
+/*
+	Orchestrates the tools required to retrieve a pokemon by ID.
+*/
+func (s PokemonService) GetPokemonByID(key uint64) (*model.Pokemon, error) {
+	if key > TOTAL_POKEMON {
+		return nil, errors.DomainValidationError{
+			Message: "OUT OF BOUND, CURRENTLY THERE ARE ONLY " + strconv.Itoa(TOTAL_POKEMON) + " POKEMON",
+		}
+	}
+	p, err := s.CSVPokemonRepository.FindByID(key)
```

Comment:
Assignments should be only cuddled with other assignments; it gives better readability to your code and separates distinct functionality code blocks. Take it in mind for future deliveries.

```suggestion
	}
	
	p, err := s.CSVPokemonRepository.FindByID(key)
```